File: 涡虫基因组-nature25473/代码/stat.fasta.length.py
#! /usr/bin/env python
import sys
if len(sys.argv) < 2:
	print (f"python {sys.argv[0]} infa > outfile")
	sys.exit()
seq={}
with open(sys.argv[1],'r') as f:
	for line in f.readlines():
		if line.startswith('>'):
			seqName = line.replace('\n', '')
			seq[seqName]=[]
		else:
			seq[seqName].append(line.strip('\n'))

for k, v in seq.items():
	seq[k] = ''.join(v)
	leng = len(seq[k])
	k = k.replace('>','').split(' ')[0];
	len2 = "{:.2f}".format(leng/10**6); 
	print (f"{k}\t{leng}\t{len2}")	

File: 涡虫基因组-nature25473/代码/stat.fasta.length.pl
#! /usr/bin/perl -w
use strict;
die "perl  $0 infa > outfile\n" unless @ARGV == 1;
open IN,$ARGV[0] || die;
$/=">";<IN>;$/="\n";
while(<IN>){
        chomp;
        my $id = (split /\s+/,$_)[0];
	$id =~ s/^>//;
        $/=">";chomp(my $fa=<IN>);$/="\n";
        $fa =~ s/\n+//g;
        my $len = length $fa;
	my $len2 = sprintf "%.2f",$len/10**6;
	print join ("\t",$id,$len,$len2),"\n";
}
close IN;
